networkx/algorithms/centrality/closeness.py

```diff
@@ -116,16 +119,156 @@ def closeness_centrality(G, u=None, distance=None, wf_improved=True):
     closeness_centrality = {}
     for n in nodes:
         sp = dict(path_length(G, n))
-        totsp = sum(sp.values())
-        if totsp > 0.0 and len(G) > 1:
-            closeness_centrality[n] = (len(sp) - 1.0) / totsp
-            # normalize to number of nodes-1 in connected part
-            if wf_improved:
-                s = (len(sp) - 1.0) / (len(G) - 1)
-                closeness_centrality[n] *= s
-        else:
-            closeness_centrality[n] = 0.0
+        closeness_centrality[n] = _node_closeness_centrality(
+            len(G), sp, wf_improved)
     if u is not None:
         return closeness_centrality[u]
     else:
         return closeness_centrality
+
+
+@not_implemented_for('directed')
+def incremental_closeness_centrality(G,
+                                     edge,
+                                     prev_cc=None,
+                                     insertion=True,
+                                     wf_improved=True):
+    r"""Incremental closeness centrality for nodes.
+
+    Compute closeness centrality for nodes using level-based work filtering
```

Comment:
Can you add some words about what incremental is? You've got "edge" as an input and no description of edges anywhere in the description of the function. Actually, it would be good to re-read all the docstrings with an eye of an inexperienced user who doesn't quite know what this whole incremental closeness thing is anyway. Using a phrase like "level-based work filtering" just replaces one jargon (incremental) with another. Try to get the true essence of the measure/method down to a couple sentences that inexperienced users can understand.